tests/utils/tools.php

```diff
@@ -243,13 +243,13 @@ function throws($function, $exceptionname, $infunction = null) {
                 if (strcasecmp($function, $infunction) == 0) {
                     printf("OK: Got %s thrown from %s\n", $exceptionname, $infunction);
                 } else {
-                    printf("ALMOST: Got %s - but was thrown in %s, not %s\n", $exceptionname, $function, $infunction);
+                    printf("ALMOST: Got %s - but was thrown in %s, not %s (%s)\n", $exceptionname, $function, $infunction, $e->getMessage());
                 }
                 return $e->getMessage();
             }
             printf("OK: Got %s\n", $exceptionname);
         } else {
-            printf("ALMOST: Got %s - expected %s\n", get_class($e), $exceptionname);
+            printf("ALMOST: Got %s - expected %s: %s\n", get_class($e), $exceptionname, $e->getMessage());
```

Comment:
@bjori: Are you asking for an [`rtrim()`](http://php.net/rtrim) (or `rtrim("\r\n")`) call on the exception message parameters to `printf()`? If so, I can make that change tomorrow morning and merge this.